### DISCUSSION TITLE
Question: Interop with Ibis?

### DISCUSSION THREAD

**Discussion by NickCrews** [perspective-dev/perspective#2673] [Q&A] [answered]
### Description of Problem:

I want to be able to use perspective as a EDA tool for [ibis](https://ibis-project.org/) tables. I want to be able to use it from vscode notebook editor with very minimal user friction. I want it to work with remote datasets, ie without materializing the entire (potentially huge) DB table to arrow.

### Potential Solutions:

Ideally I would love something like

```python
import perspective_ibis as pi
import ibis

t = ibis.examples.penguins.fetch()
pi.show(t)

# ideally, could even monkeypatch Ibis's __repr__s
perspective_ibis.install_repr()
t
```

A jupyterwidget, possibly built with [anywidget](https://github.com/manzt/anywidget), seems like it may provide a nice two way binding from python to js.

I don't really understand the data model of perspective though, not sure if this is possible. It looks like you hand perspective ALL of your data via arrow, and then it is responsible for filter/sort/groupby/etc. But the point of ibis is to be able to work with an eg 100GB table in bigquery without actually pulling down the whole thing to your machine. Instead, I think what we would need to do is have some sort of hook in python that takes the json spec from perspective, parses it, uses ibis to transform the input table, materializes it to arrow, and then passes this back to perspective?

Thank you!

**Comment by texodus** [comment, marked as answer]
> A jupyterwidget, possibly built with [anywidget](https://github.com/manzt/anywidget), seems like it may provide a nice two way binding from python to js.

[`PerspectiveWidget`](https://perspective.finos.org/docs/python/#perspectivewidget)

> I don't really understand the data model of perspective though, not sure if this is possible. It looks like you hand perspective ALL of your data via arrow, and then it is responsible for filter/sort/groupby/etc.

[I've already answered this here](https://github.com/finos/perspective/discussions/1361#discussioncomment-5285191).
